front/lib/providers.ts

```diff
@@ -104,14 +112,23 @@ export const serviceProviders: ServiceProvider[] = [
 export async function checkProvider(
   owner: WorkspaceType,
   providerId: string,
-  config: object
+  config: object,
+  usePost = false
 ): Promise<GetProvidersCheckResponseBody> {
   try {
-    const result = await fetch(
-      `/api/w/${
-        owner.sId
-      }/providers/${providerId}/check?config=${JSON.stringify(config)}`
```

Comment:
Let's simply base64 encode it?

Reply:
It's a really bad security practice to send sensitive stuff through the query string, I'm not really comfortable doing it with a service account

Reply:
If anything I'd rather move every other check to POST

Reply:
As per slack, merging and moving other checks to POST